cmd/testdata/success_nestjs.golden

```diff
@@ -0,0 +1,194 @@
+}
+
+/**
+ * Options for injecting an OpenFeature client into a constructor.
+ */
+interface FeatureClientProps {
+  /**
+   * The domain of the OpenFeature client, if a domain scoped client should be used.
+   * @see {@link Client.getBooleanDetails}
+   */
+  domain?: string;
+}
+
+/**
+ * Injects a generated typesafe feature client into a constructor or property of a class.
+ * @param {FeatureClientProps} [props] The options for injecting the client.
+ * @returns {PropertyDecorator & ParameterDecorator} The decorator function.
+ */
+export const GeneratedOpenFeatureClient = (props?: FeatureClientProps): PropertyDecorator & ParameterDecorator =>
+  Inject(getOpenFeatureGeneratedClientToken(props?.domain));
+
+/**
+ * GeneratedOpenFeatureModule is a generated typesafe NestJS wrapper for OpenFeature Server-SDK.
+ */
+@Module({})
+export class GeneratedOpenFeatureModule extends OpenFeatureModule {
+  static override forRoot({ useGlobalInterceptor = true, ...options }: OpenFeatureModuleOptions): DynamicModule {
+    const module = super.forRoot({ useGlobalInterceptor, ...options });
+
+    const clientValueProviders: NestFactoryProvider<GeneratedClient>[] = [
+      {
+        provide: getOpenFeatureGeneratedClientToken(),
+        useFactory: () => getGeneratedClient(),
+      },
+    ];
+
+    if (options?.providers) {
+      const domainClientProviders: NestFactoryProvider<GeneratedClient>[] = Object.keys(options.providers).map(
+        (domain) => ({
+          provide: getOpenFeatureGeneratedClientToken(domain),
+          useFactory: () => getGeneratedClient(domain),
+        }),
+      );
+
+      clientValueProviders.push(...domainClientProviders);
+    }
+
+    return {
+      ...module,
+      providers: module.providers ? [...module.providers, ...clientValueProviders] : clientValueProviders,
+      exports: module.exports ? [...module.exports, ...clientValueProviders] : clientValueProviders,
+    };
+  }
+}
+
+/**
+ * Options for injecting a typed feature flag into a route handler.
+ */
+interface TypedFeatureProps {
+  /**
+   * The domain of the OpenFeature client, if a domain scoped client should be used.
+   * @see {@link OpenFeature#getClient}
+   */
+  domain?: string;
+  /**
+   * The {@link EvaluationContext} for evaluating the feature flag.
+   * @see {@link OpenFeature#getClient}
+   */
+  context?: EvaluationContext;
+}
+
+
+/**
+ * Gets the {@link EvaluationDetails} for `discountPercentage` from a domain scoped or the default OpenFeature
+ * client and populates the annotated parameter with the {@link EvaluationDetails} wrapped in an {@link Observable}.
+ *
+ * **Details:**
+ * - flag key: `discountPercentage`
+ * - description: `Discount percentage applied to purchases.`
+ * - default value: `0.15`
+ * - type: `number`
+ *
+ * Usage:
+ * ```typescript
+ * @Get("/")
+ * public async handleRequest(
+ *     @DiscountPercentage()
+ *     discountPercentage: Observable<EvaluationDetails<number>>,
+ * )
+ * ```
```

Comment:
I like the usage example! I'll probably borrow this idea for the React and node generators.